sunpy/net/dataretriever/sources/tests/test_gong_synoptic.py

```diff
@@ -0,0 +1,112 @@
+     'https://gong2.nso.edu/oQR/zqs/202004/mrzqs200421/mrzqs200421t2314c2230_335.fits.gz'
+     ),
+    (TimeRange('2006/9/19', '2006/9/19 22:00'),
+     'https://gong2.nso.edu/oQR/zqs/200609/mrzqs060919/mrzqs060919t1154c2048_323.fits.gz',
+     'https://gong2.nso.edu/oQR/zqs/200609/mrzqs060919/mrzqs060919t1754c2048_320.fits.gz'
+     )
+])
+def test_get_url_for_time_range(GSClient, timerange, url_start, url_end):
+    urls = GSClient._get_url_for_timerange(timerange)
+    assert isinstance(urls, list)
+    assert urls[0] == url_start
+    assert urls[-1] == url_end
+
+
+@given(range_time('2010-01-06'))
+def test_can_handle_query(time):
+    ans1 = gong_synoptic.GongSynopticClient._can_handle_query(
+        time, Instrument('gong'))
+    assert ans1 is True
+    ans2 = gong_synoptic.GongSynopticClient._can_handle_query(time)
+    assert ans2 is False
+    ans3 = gong_synoptic.GongSynopticClient._can_handle_query(time, Instrument('goes'))
+    assert ans3 is False
+    ans4 = gong_synoptic.GongSynopticClient._can_handle_query(
+        time, Instrument('gong'), a.Physobs('LOS_MAGNETIC_FIELD'),
+        a.gong_synoptic.ExtentType('synoptic'))
+    assert ans4 is True
+    ans4 = gong_synoptic.GongSynopticClient._can_handle_query(
+        time, Instrument('gong'), a.Physobs('LOS_MAGNETIC_FIELD'),
+        a.gong_synoptic.ExtentType('synoptic'))
+    assert ans4 is True
+    ans5 = gong_synoptic.GongSynopticClient._can_handle_query(
+        time, Instrument('gong'), a.gong_synoptic.ExtentType('synoptic'))
+    assert ans5 is True
+    ans6 = gong_synoptic.GongSynopticClient._can_handle_query(
+        time, Instrument('gong'), a.gong_synoptic.ExtentType('FULL_DISK'))
+    assert ans6 is False
+
+
+@pytest.mark.parametrize("time", [
+    Time('2015/8/27', '2015/8/27 12:00'),
+    Time('2020/2/4', '2020/2/5')])
+@pytest.mark.remote_data
+def test_query(GSClient, time):
+    qr1 = GSClient.search(time, Instrument('gong'))
+    assert isinstance(qr1, QueryResponse)
+    assert qr1.time_range().start == time.start
+    assert qr1.time_range().end == time.end
+
+
+@pytest.mark.remote_data
+@pytest.mark.parametrize("time,instrument", [
+    (Time('2013/8/27', '2013/8/27'), Instrument('gong')),
+    (Time('2020/4/23 17:00', '2020/4/23 21:00'), Instrument('gong')),
+])
+def test_get(GSClient, time, instrument):
+    qr1 = GSClient.search(time, instrument)
+    download_list = GSClient.fetch(qr1)
+    assert len(download_list) == len(qr1)
+
+
+@pytest.mark.remote_data
+@pytest.mark.parametrize(
+    "time, instrument",
+    [(a.Time('2019/10/4', '2019/10/4 2:00'), a.Instrument('gong')),
+     (a.Time('2019/12/31 21:00', '2020/1/1'), a.Instrument('gong'))])
+def test_fido(time, instrument):
+    qr = Fido.search(time, instrument)
+    assert isinstance(qr, UnifiedResponse)
+    response = Fido.fetch(qr)
+    assert len(response) == qr._numfile
+    for res in response:
+        assert res.split('.')[-1] == 'gz'
```

Comment:
I don't really like `assert` on loops, if one fails, you only get a "report" of that one, and not whether others did too.
```suggestion
    assert all(map(lambda x: x.endswith('.gz'), response))
```